components/infobox/commons/infobox_league.lua

```diff
@@ -429,8 +429,10 @@ function League:_cleanDate(date)
 		return nil
 	end
 
-	date = date:gsub('-??', '-01')
-	date = date:gsub('-XX', '-01')
+	-- due to '-' and '?' being part of the 'magic' characters for patterns
+	-- we have to escape them with '%'
+	date = date:gsub('%-%?%?', '-01')
+	date = date:gsub('%-XX', '-01')
```

Comment:
Shouldn't this just use ReferenceCleaner?

Reply:
feel free to adjust, i am on mobile until Friday, so i will not do it